app/uk/gov/hmrc/bindingtariffclassification/scheduler/ReferredDaysElapsedJob.scala

```diff
@@ -0,0 +1,128 @@
+class ReferredDaysElapsedJob @Inject()(appConfig: AppConfig,
+                                       caseService: CaseService,
+                                       eventService: EventService,
+                                       bankHolidaysConnector: BankHolidaysConnector) extends ScheduledJob {
+
+  private implicit val carrier: HeaderCarrier = HeaderCarrier()
+  private lazy val jobConfig = appConfig.referredDaysElapsed
+  private lazy val criteria = CaseSearch(
+    filter = CaseFilter(statuses = Some(Set(PseudoCaseStatus.REFERRED))),
+    sort = Some(CaseSort(CaseSortField.REFERENCE))
+  )
+
+  override val name: String = "ReferredDaysElapsed"
+
+  override def interval: FiniteDuration = jobConfig.interval
+
+  override def firstRunTime: LocalTime = jobConfig.elapseTime
+
+  override def execute(): Future[Unit] = for {
+    bankHolidays <- bankHolidaysConnector.get()
+    _ <- process(1)(bankHolidays)
+  } yield ()
+
+  private def process(page: Int)(implicit bankHolidays: Set[LocalDate]): Future[Unit] = {
+    caseService.get(criteria, Pagination(page = page)) flatMap { pager =>
+      sequence(pager.results.map(refreshDaysElapsed)).map(_ => pager)
+    } flatMap {
+      case pager if pager.hasNextPage => process(page + 1)
+      case _ => successful(())
+    }
+  }
+
+  private def refreshDaysElapsed(c: Case)(implicit bankHolidays: Set[LocalDate]): Future[Unit] = {
+    val createdDate: LocalDate = LocalDateTime.ofInstant(c.createdDate, ZoneOffset.UTC).toLocalDate
+    val daysSinceCreated: Long = ChronoUnit.DAYS.between(createdDate, LocalDate.now(appConfig.clock))
+
+    // Working days between the created date and Now
+    val workingDays: Seq[Instant] = (0L until daysSinceCreated)
+      .map(createdDate.plusDays)
+      .filterNot(bankHoliday)
+      .filterNot(weekend)
+      .map(toInstant)
+
+    for {
+      // Get the Status Change events for that case
+      events <- eventService.search(EventSearch(Some(Set(c.reference)), Some(Set(EventType.CASE_STATUS_CHANGE))), Pagination(1, Integer.MAX_VALUE))
+
+      // Generate a timeline of the Case Status over time
+      statusTimeline: StatusTimeline = StatusTimeline.from(events.results)
+
+      // Filter down to the days the case was Referred
+      referredDays: Seq[Instant] = workingDays
+        .filter(statusTimeline.statusOn(_).contains(CaseStatus.REFERRED))
+
+      // Update the case
+      _ <- caseService.update(c.copy(referredDaysElapsed = referredDays.size), upsert = false)
+
+      _ = Logger.info(s"$name: Updated Days Elapsed of Case [${c.reference}] from [${c.daysElapsed}] to [${referredDays.size}]")
+    } yield ()
+  }
+
+  private def toInstant: LocalDate => Instant = _.atStartOfDay(appConfig.clock.getZone).toInstant
+
+  private def bankHoliday(date: LocalDate)(implicit bankHolidays: Set[LocalDate]): Boolean = bankHolidays.contains(date)
+
+  private def weekend(date: LocalDate): Boolean = Set(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY).contains(date.getDayOfWeek)
```

Comment:
These three methods are duplicated in the existing job are they not?  Could we not extract somewhere

Reply:
They could potentially go into the Scheduler Date Util